src/include/platform/internal/GenericDeviceInstanceInfoProvider.ipp

```diff
@@ -21,6 +21,20 @@ namespace chip {
 namespace DeviceLayer {
 namespace Internal {
 
+template <class ConfigClass>
+CHIP_ERROR GenericDeviceInstanceInfoProvider<ConfigClass>::GetDeviceName(MutableCharSpan & deviceNameSpan)
+{
+
+    constexpr char deviceName[] = CHIP_DEVICE_CONFIG_DEVICE_NAME;
+
+    ReturnErrorCodeIf(sizeof(deviceName) > deviceNameSpan.size(), CHIP_ERROR_BUFFER_TOO_SMALL);
+    ReturnErrorCodeIf(sizeof(deviceName) != ConfigurationManager::kMaxDeviceNameLength, CHIP_ERROR_BUFFER_TOO_SMALL);
```

Comment:
Why are we requiring that CHIP_DEVICE_CONFIG_DEVICE_NAME be exactly 64 chars long?

Reply:
Returned it back to 32 characters long

Reply:
and removed the != check